### DISCUSSION TITLE
DsfrNavigationMenu : le bouton Drop down n'apparait pas actif

### DISCUSSION THREAD

**Discussion by ChrisD92** [dnum-mi/vue-dsfr#596] [Q&A]
Bonjour, 

Tout d'abord, merci pour ce portage du DSFR bien pratique.

Je cherche à utiliser le menu déroulant DsfrNavigationMenu. Mais lorsque je navigue sur une page en cliquant sur un lien du menu, le menu se ferme (fonctionnement normal) mais le bouton drop down n'est pas marqué comme actif pour signaler que l'on se situe bien sur une page du menu en question.

Mon code : 

Script

`const navItems = ref(
{
  title: 'Nom de menu original',
  links: [
    {
      text: 'Valeur 1',
      to: '/lien1',
    },
    {
      text: 'Valeur 2',
      to: '/lien2',
    },
    {
      text: 'Valeur 3',
      to: '/lien3',
    },
  ],
}

const expandedMenuId = ref(undefined)
const toggle = (event:any) => {
  expandedMenuId.value = event
}
`

Template :
`
 <dsfr-navigation>

        <dsfr-navigation-item>
          <dsfr-navigation-menu-link to="/" text="Accueil" />
        </dsfr-navigation-item>

        <dsfr-navigation-item>
          <dsfr-navigation-menu
            :title="navItems.title"
            :links="navItems.links"
            :expanded-id="expandedMenuId"
            @toggle-id="toggle($event)"
          />
        </dsfr-navigation-item>

        <dsfr-navigation-item>
          <dsfr-navigation-menu-link to="/articles" text="Actualités" />
        </dsfr-navigation-item>
      </dsfr-navigation>

`

j'ai l'impression que le bouton drop down n'a pas l'attribut aria-current="true" dans le html généré

Est-ce que vous pouvez m'aider ?

**Comment by laruiss** [comment]
Bonjour,

Si vous n’utilisez pas directement et uniquement DsfrNavigation, c’est à vous de donner un `id` à chaque menu et sous-menu et liens et de donner le `expandedId` à chacun d’eux.

**Reply by ChrisD92** [reply]
C'est le même problème en utilisant uniquement que `<dsfr-navigation :nav-items="navItems">`
et navItems dans un Array

**Comment by ChrisD92** [comment]
`<button class="fr-nav__btn" aria-expanded="false" aria-controls="menu-soda0" data-fr-js-collapse-button="true"><span>Titre bouton</span></button>`

Html du bouton généré lorsque je suis bien sur une page d'un des liens du menu

**Reply by ChrisD92** [reply]
Et j'ai l'impression que c'est le même problème avec le composant DsfrSideMenu pour les drop down. Il met bien actif le lien à l'intérieur du menu mais pas ses boutons parents

**Reply by laruiss** [reply]
Bonjour, et merci pour votre patience. Le problème est simple, c’est l’attribut `aria-current` qui est manquant. La solution est malheureusement plus compliquée. J’en ai trouvé une assez satisfaisante (je l’espère). #634

**Reply by laruiss** [reply]
En fait ça n’ira pas, parce que ce n’est pas SSR-ready. Donc je vais faire autrement.
